src/ser/mod.rs

```diff
@@ -897,6 +897,28 @@ mod tests {
 
     use serde_derive::Deserialize;
 
+    #[test]
+    fn serialize_nested_unit() {
+        #[derive(Debug, Deserialize, Serialize, PartialEq)]
+        #[serde(rename_all = "lowercase")]
+        pub enum UnitResult {
+            Ok(()),
+            Err(String),
+        }
+
+        let err_input = UnitResult::Err("some error".to_string());
+        let json = to_string(&err_input).expect("encode err enum");
+        assert_eq!(json, r#"{"err":"some error"}"#.to_string());
+        let loaded = crate::from_str(&json).expect("re-load err enum");
+        assert_eq!(err_input, loaded);
+
+        let ok_unit = UnitResult::Ok(());
+        let json = to_string(&ok_unit).expect("encode ok enum");
+        assert_eq!(json, r#"{"ok":null}"#.to_string());
+        let loaded = crate::from_str(&json).expect("re-load ok enum");
+        assert_eq!(ok_unit, loaded);
+    }
```

Comment:
Looks good. If you like this could even be integrated into `serialize_embedded_enum` below if we use a 3rd enum case.